<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title>Finding differentially expressed microRNA in the  AHUS and UCAM data sets</title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<h1>Finding differentially expressed microRNA in the  AHUS and UCAM data sets</h1>

<p>2015-06-23 18:38:15</p>

<p><br/>
<br/></p>

<h2>Introduction</h2>

<p>The aim of the analysis performed in this report is to validate some of the findings from <a href="http://www.pnas.org/content/early/2012/02/01/1200010109">Volinia et al.</a>, 
based on two independent data sets using 3 different Agilent microRNA microarrays from two providers called <strong>AHUS</strong> and  <strong>UCAM</strong>.</p>

<p>More specifically:</p>

<ul>
<li>Combine the data into one big sample vs. microRNA matrix</li>
<li>Do a &ldquo;find differentially expressed genes&rdquo;- test between the tissue types DCIS-normal and invasive-DCIS</li>
<li>With a significance cut-off, report the overlap with what is given in Volinia et al. <a href="http://www.pnas.org/content/suppl/2012/02/02/1200010109.DCSupplemental/pnas.201200010SI.pdf">(Suppl. table S1 and S2)</a> </li>
<li>Do an additional  &ldquo;find differentially expressed genes&rdquo;- test between the tissue types DCIS and the individual pam50 subtypes based on classification performed on mRNA samples.</li>
</ul>

<p>This report is made with RStudio and knitr in order to tie the description, code, plots and results together, and hopefully to limit confusion and adhere to a more reproducible research style. </p>

<p><br/>
<br/></p>

<h2>Input data</h2>

<p>Setting some dependencies</p>

<pre><code class="r">library(limma)
library(xtable)
library(knitr)

set.seed(100)
if(!exists(&quot;inputisread&quot;))
    source(&quot;read_input.r&quot;)

outputdir = paste( &quot;output-analysis&quot;, sep=&quot;&quot;)
if(!file.exists(outputdir))
    dir.create(outputdir)
difflistdir = paste(outputdir, &quot;/difflists_merged&quot;, sep=&quot;&quot;)
if(!file.exists(difflistdir))
    dir.create(difflistdir)
postfix = &quot;.txt&quot;
</code></pre>

<p>The datamatrix has 1573 samples and 265 microRNAs.</p>

<pre><code class="r">print(xtable(table(sampleannotation[, c(&quot;provider&quot;, &quot;tissue_type&quot;)]), 
             caption=&quot;&quot;, digits=3), 
      comment = TRUE,
      type = &quot;html&quot;,
      html.table.attributes=&quot;CELLPADDING=5&quot;,
      include.rownames = TRUE)
</code></pre>

<!-- html table generated in R 3.1.1 by xtable 1.7-4 package -->

<!-- Tue Jun 23 18:38:15 2015 -->

<table CELLPADDING=5>
<caption align="bottom">  </caption>
<tr> <th>  </th> <th> benign </th> <th> DCIS </th> <th> invasive </th> <th> normal </th>  </tr>
  <tr> <td align="right"> AHUS </td> <td align="right">   23 </td> <td align="right">    8 </td> <td align="right">   55 </td> <td align="right">   70 </td> </tr>
  <tr> <td align="right"> UCAM </td> <td align="right">    8 </td> <td align="right">   10 </td> <td align="right"> 1283 </td> <td align="right">  116 </td> </tr>
   </table>

<p>The tissue is not evenly distributed among the providers. &ldquo;benign&rdquo; is not part of Volinia et al.s results, but it is kept for now since it is useful in assessing batch differences. </p>

<p><br/>
<br/></p>

<h2>Finding differetially expressed microRNAs</h2>

<p>Limma is used and both batch and group is included in the analysis.</p>

<p>DCIS vs normal and invasive vs. DCIS comparisons.</p>

<pre><code class="r"># setting output folder

#Limma blocked batch and significance test
group = factor(sampleannotation$tissue_type)
batch = factor(sampleannotation$provider)

#  a = batch==&quot;AHUS&quot; &amp; group %in% c(&quot;DCIS&quot;, &quot;invasive&quot;)
#  group[a] = sample(group[a])
#  b = batch==&quot;UCAM&quot; &amp; group %in% c(&quot;DCIS&quot;, &quot;invasive&quot;)
#  group[b] = sample(group[b])

design = model.matrix(~0+group+batch)
colnames(design) = gsub(&quot;group&quot;, &quot;&quot;, colnames(design))
fit = lmFit(common_matrix, design)
fit$genes$name = miRNA2MIMAT[rownames(common_matrix), &quot;preferredname&quot;]
cont.matrix = makeContrasts ( contrasts=c(&quot;DCIS-normal&quot;, &quot;invasive-DCIS&quot;), levels=design)  
fit2 = contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2)

DCIS_vs_normal_test = topTable(fit2, coef=&quot;DCIS-normal&quot;, adjust=&quot;BH&quot;, number=999)
print( paste(&quot;Diff genes with fdr&lt;0.05 for DCIS vs. normal: &quot;,
               sum(DCIS_vs_normal_test$adj.P.Val &lt; 0.05), sep=&quot;&quot;))
</code></pre>

<p>[1] &ldquo;Diff genes with fdr&lt;0.05 for DCIS vs. normal: 86&rdquo;</p>

<pre><code class="r">DCIS_vs_normal_difflistfile = paste(difflistdir, 
                                    &quot;/difflist-DCIS-vs-normal&quot;,
                                    postfix, sep=&quot;&quot;)
write.table(DCIS_vs_normal_test, 
            file=DCIS_vs_normal_difflistfile,
            quote=FALSE, sep=&quot;\t&quot;, row.names=TRUE, col.names=NA)
print(xtable(head(DCIS_vs_normal_test), caption=&quot;Top of the result from the DCIS vs normal test&quot;, digits=3), 
      comment = FALSE,
      type = &quot;html&quot;,
      html.table.attributes=&quot;CELLPADDING=5&quot;,
      include.rownames = FALSE)
</code></pre>

<table CELLPADDING=5>
<caption align="bottom"> Top of the result from the DCIS vs normal test </caption>
<tr> <th> name </th> <th> logFC </th> <th> AveExpr </th> <th> t </th> <th> P.Value </th> <th> adj.P.Val </th> <th> B </th>  </tr>
  <tr> <td> hsa-miR-1305 </td> <td align="right"> 0.681 </td> <td align="right"> 6.564 </td> <td align="right"> 13.002 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 72.622 </td> </tr>
  <tr> <td> hsa-miR-1288-3p </td> <td align="right"> 0.342 </td> <td align="right"> 6.256 </td> <td align="right"> 12.175 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 63.193 </td> </tr>
  <tr> <td> hsa-miR-21-5p </td> <td align="right"> 2.407 </td> <td align="right"> 14.925 </td> <td align="right"> 10.902 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 49.706 </td> </tr>
  <tr> <td> hsa-miR-139-5p </td> <td align="right"> -1.276 </td> <td align="right"> 7.033 </td> <td align="right"> -9.693 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 38.080 </td> </tr>
  <tr> <td> hsa-miR-1274b </td> <td align="right"> 1.600 </td> <td align="right"> 11.862 </td> <td align="right"> 9.690 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 38.054 </td> </tr>
  <tr> <td> hsa-miR-486-5p </td> <td align="right"> -1.820 </td> <td align="right"> 7.517 </td> <td align="right"> -7.843 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 22.674 </td> </tr>
   </table>

<p>Many microRNAs seems to differ between DCIS and normal.</p>

<p>Now, the invasive vs. DCIS comparison.</p>

<pre><code class="r">invasive_vs_DCIS_test =  topTable(fit2, coef=&quot;invasive-DCIS&quot;, adjust=&quot;BH&quot;, number=999)
  print( paste(&quot;Diff genes with fdr&lt;0.05 for invasive vs. DCIS: &quot;,
               sum(invasive_vs_DCIS_test$adj.P.Val &lt; 0.05), sep=&quot;&quot;))
</code></pre>

<p>[1] &ldquo;Diff genes with fdr&lt;0.05 for invasive vs. DCIS: 10&rdquo;</p>

<pre><code class="r">invasive_vs_DCIS_difflistfile=paste(difflistdir, 
                                    &quot;/difflist-invasive-vs-DCIS&quot;,
                                    postfix, sep=&quot;&quot;) 
write.table(invasive_vs_DCIS_test, 
            file=invasive_vs_DCIS_difflistfile,
            quote=FALSE, sep=&quot;\t&quot;, row.names=TRUE, col.names=NA)
print(xtable(head(invasive_vs_DCIS_test), caption=&quot;Top of the result from the invasive vs DCIS test&quot;, digits=3), 
      comment = FALSE,
      type = &quot;html&quot;,
      html.table.attributes=&quot;CELLPADDING=5&quot;,
      include.rownames = FALSE)
</code></pre>

<table CELLPADDING=5>
<caption align="bottom"> Top of the result from the invasive vs DCIS test </caption>
<tr> <th> name </th> <th> logFC </th> <th> AveExpr </th> <th> t </th> <th> P.Value </th> <th> adj.P.Val </th> <th> B </th>  </tr>
  <tr> <td> hsa-miR-1288-3p </td> <td align="right"> -0.302 </td> <td align="right"> 6.256 </td> <td align="right"> -11.059 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 51.366 </td> </tr>
  <tr> <td> hsa-miR-1305 </td> <td align="right"> -0.546 </td> <td align="right"> 6.564 </td> <td align="right"> -10.704 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 47.789 </td> </tr>
  <tr> <td> hsa-miR-1274b </td> <td align="right"> -0.931 </td> <td align="right"> 11.862 </td> <td align="right"> -5.791 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 9.256 </td> </tr>
  <tr> <td> hsa-miR-15b-5p </td> <td align="right"> 0.715 </td> <td align="right"> 12.638 </td> <td align="right"> 4.750 </td> <td align="right"> 0.000 </td> <td align="right"> 0.000 </td> <td align="right"> 3.889 </td> </tr>
  <tr> <td> hsa-miR-720 </td> <td align="right"> -0.793 </td> <td align="right"> 14.486 </td> <td align="right"> -4.177 </td> <td align="right"> 0.000 </td> <td align="right"> 0.002 </td> <td align="right"> 1.371 </td> </tr>
  <tr> <td> hsa-miR-107 </td> <td align="right"> 0.588 </td> <td align="right"> 10.031 </td> <td align="right"> 3.963 </td> <td align="right"> 0.000 </td> <td align="right"> 0.003 </td> <td align="right"> 0.514 </td> </tr>
   </table>

<h2>Stratification based on pam50 classification.</h2>

<p>Pam50 classification has been done on the mRNA samples. Many of the microRNA samples also have a mRNA sample and thus a pam50 classification. </p>

<pre><code class="r">table(sampleannotation[,c(&quot;tissue_type&quot;, &quot;pam50_est&quot;)], useNA=&quot;ifany&quot;)
</code></pre>

<pre><code>##            pam50_est
## tissue_type Basallike DCIS Her2 LumA LumB Normallike unknown
##    benign           2    0    0    3    0         12      14
##    DCIS             0   18    0    0    0          0       0
##    invasive       198    0  167  476  385         97      15
##    normal           0    0    0    9    1        105      71
</code></pre>

<p>We have too few samples for looking at the DCIS to invasive change for each of the pam50 classes. We will treat all DCIS as &ldquo;DCIS&rdquo;.</p>

<pre><code class="r">pam50 = sampleannotation$pam50_est
pam50[sampleannotation$tissue_type==&quot;DCIS&quot;]=&quot;DCIS&quot;
pam50labels = unique(pam50)
pam50labels = pam50labels[!pam50labels %in% c(&quot;unknown&quot;, &quot;DCIS&quot;)]
</code></pre>

<p>The diff test,</p>

<pre><code class="r">pam50contrasts = paste(pam50labels, &quot;-DCIS&quot;, sep=&quot;&quot;)
to_be_used = sampleannotation$tissue_type %in% c(&quot;DCIS&quot;, &quot;invasive&quot;) &amp; (pam50 != &quot;unknown&quot;) # do not use samples without clinical info!

group = factor(pam50[to_be_used])
batch = factor(sampleannotation$provider[to_be_used])

# simple permutation sanity check
#a = batch==&quot;AHUS&quot; &amp; group %in% c(&quot;DCIS&quot;, pam50labels)
#group[a] = sample(group[a])
#b = batch==&quot;UCAM&quot; &amp; group %in% c(&quot;DCIS&quot;, pam50labels)
#group[b] = sample(group[b])

design = model.matrix(~0+group+batch)
colnames(design) = gsub(&quot;group&quot;, &quot;&quot;, colnames(design))
fit = lmFit(common_matrix[,to_be_used], design)
fit$genes$name = miRNA2MIMAT[rownames(common_matrix), &quot;preferredname&quot;]
cont.matrix = makeContrasts ( contrasts=pam50contrasts, levels=design)  
fit2 = contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2)

for(thiscontrast in pam50contrasts)
{
  thisres = topTable(fit2, coef=thiscontrast, adjust=&quot;BH&quot;, number=999)

  thisname= gsub(&quot;-&quot;, &quot;-vs-&quot;, thiscontrast)
  thislabel = strsplit(thiscontrast, &quot;-&quot;)[[1]][1]

  difflistfile=paste(difflistdir, &quot;/difflist-pam50-&quot;,thisname, postfix, sep=&quot;&quot;)
  write.table(thisres, 
            file=difflistfile,
            quote=FALSE, sep=&quot;\t&quot;, row.names=TRUE, col.names=NA)
  print( paste(&quot;Diff genes with fdr&lt;0.05 for &quot;, thislabel, &quot; vs DCIS: &quot;,
               sum(thisres$adj.P.Val &lt; 0.05), sep=&quot;&quot;))

}
</code></pre>

<pre><code>## [1] &quot;Diff genes with fdr&lt;0.05 for Normallike vs DCIS: 37&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for LumA vs DCIS: 31&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for LumB vs DCIS: 43&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for Her2 vs DCIS: 16&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for Basallike vs DCIS: 61&quot;
</code></pre>

<p>This is diff genes between all the DCIS (which is a small and diverse group) and each of the pam50 classes for the invasive samples.</p>

<p><br/>
<br/></p>

<h2>Stratification based on Immunohistochemistry (IHC) classification.</h2>

<p>Many of the samples are also diagnosed based on immunohistochemistry of HER2, PGR and ER proteins. Four diagnoses are used, 
&ldquo;HER2neg/ERneg/PGRneg&rdquo;, &ldquo;HER2neg/ERpos&rdquo;, &ldquo;HER2pos/ERneg&rdquo;, &ldquo;HER2pos/ERpos&rdquo;. 
We would like too look for microRNA that are changed between all the DCIS and the different IHC diagnoses of invasive samples.</p>

<pre><code class="r">IHC = sampleannotation$IHC
IHC[sampleannotation$tissue_type==&quot;DCIS&quot;] = &quot;DCIS&quot;
table(sampleannotation[,&quot;tissue_type&quot;], IHC, useNA=&quot;ifany&quot;)
</code></pre>

<pre><code>##           IHC
##            DCIS HER2neg_ERneg_PGRneg HER2neg_ERpos HER2pos_ERneg
##   benign      0                    5             2             0
##   DCIS       18                    0             0             0
##   invasive    0                  199           940            87
##   normal      0                    0             0             0
##           IHC
##            HER2pos_ERpos unknown
##   benign               0      24
##   DCIS                 0       0
##   invasive            93      19
##   normal               0     186
</code></pre>

<p>The diff test,</p>

<pre><code class="r">IHClabels = unique(IHC[!IHC %in% c(&quot;&quot;, &quot;unknown&quot;, &quot;DCIS&quot;)])
IHCcontrasts = paste(IHClabels, &quot;-DCIS&quot;, sep=&quot;&quot;)
to_be_used =  sampleannotation$tissue_type %in% c(&quot;DCIS&quot;, &quot;invasive&quot;) &amp; (IHC != &quot;unknown&quot;) # do not use samples without clinical info!

group = factor(IHC[to_be_used])
batch = factor(sampleannotation$provider[to_be_used])

#simple permutation sanity check
# a = batch==&quot;AHUS&quot; &amp; group %in% c(&quot;DCIS&quot;, IHClabels)
# group[a] = sample(group[a])
# b = batch==&quot;UCAM&quot; &amp; group %in% c(&quot;DCIS&quot;, IHClabels)
# group[b] = sample(group[b])

design = model.matrix(~0+group+batch)
colnames(design) = gsub(&quot;group&quot;, &quot;&quot;, colnames(design))
fit = lmFit(common_matrix[,to_be_used], design)
fit$genes$name = miRNA2MIMAT[rownames(common_matrix), &quot;preferredname&quot;]
cont.matrix = makeContrasts ( contrasts=IHCcontrasts, levels=design)  
fit2 = contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2)

for(thiscontrast in IHCcontrasts)
{
  thisres = topTable(fit2, coef=thiscontrast, adjust=&quot;BH&quot;, number=999)

  thisname= gsub(&quot;-&quot;, &quot;-vs-&quot;, thiscontrast)
  thislabel = strsplit(thiscontrast, &quot;-&quot;)[[1]][1]

  difflistfile=paste(difflistdir, &quot;/difflist-IHC-&quot;,thisname, postfix, sep=&quot;&quot;)
  write.table(thisres, 
            file=difflistfile,
            quote=FALSE, sep=&quot;\t&quot;, row.names=TRUE, col.names=NA)
  print( paste(&quot;Diff genes with fdr&lt;0.05 for &quot;, thislabel, &quot; vs DCIS: &quot;,
               sum(thisres$adj.P.Val &lt; 0.05), sep=&quot;&quot;))

}
</code></pre>

<pre><code>## [1] &quot;Diff genes with fdr&lt;0.05 for HER2neg_ERpos vs DCIS: 11&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for HER2pos_ERneg vs DCIS: 19&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for HER2pos_ERpos vs DCIS: 5&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for HER2neg_ERneg_PGRneg vs DCIS: 30&quot;
</code></pre>

<p><strong>Important note regarding Limma results</strong>
For the diff-tests I ran a few sanity checks where I permuted the group labels whitin batches. For every 3 of 10 run I got a few genes, and once in a while I did get many more. This indicates that some assumtions of the model might be broken to a degree that the resulting p-values should be considered somewhat optimistic.</p>

<p><br/>
<br/></p>

<h2>Stratification based on iClust-subtypes</h2>

<p>The UCAM samples are classified as an iClust-subtype based on mRNA samples. This is provided as a sample annotationlike the pam50 estimates. We will finde differentially expressed microRNA between the DCIS samples and each of the iClust-types. This classification is onlye aveilible for the UCAM dataset and no alternative meta-analysis is done.</p>

<pre><code class="r">iClust = sampleannotation$iClust
iClust[!is.na(iClust)] = paste(&quot;iClust&quot;, iClust[!is.na(iClust)], sep=&quot;&quot;)
iClust[sampleannotation$tissue_type==&quot;DCIS&quot;] = &quot;DCIS&quot;
to_be_used =  sampleannotation$tissue_type %in% c(&quot;DCIS&quot;, &quot;invasive&quot;) &amp;
    !is.na(iClust) &amp; # do not use samples without clinical info
    sampleannotation$provider==&quot;UCAM&quot; 
table(iClust[to_be_used],sampleannotation[to_be_used,&quot;tissue_type&quot;], useNA=&quot;ifany&quot;)
</code></pre>

<pre><code>##           
##            DCIS invasive
##   DCIS       10        0
##   iClust1     0       91
##   iClust10    0      134
##   iClust2     0       52
##   iClust3     0      191
##   iClust4     0      242
##   iClust5     0      122
##   iClust6     0       56
##   iClust7     0      114
##   iClust8     0      186
##   iClust9     0       87
</code></pre>

<p>The diff test,</p>

<pre><code class="r">labels = unique(iClust[to_be_used &amp; !iClust %in% c(&quot;&quot;, &quot;unknown&quot;, &quot;DCIS&quot;)])
contrasts = paste(labels, &quot;-DCIS&quot;, sep=&quot;&quot;)

group = factor(iClust[to_be_used])
#group[group!=&quot;DCIS&quot;] = sample(group[group!=&quot;DCIS&quot;])#simple sanity check

design = model.matrix(~0+group)
colnames(design) = gsub(&quot;group&quot;, &quot;&quot;, colnames(design))
fit = lmFit(common_matrix[,to_be_used], design)
fit$genes$name = miRNA2MIMAT[rownames(common_matrix), &quot;preferredname&quot;]
cont.matrix = makeContrasts ( contrasts=contrasts, levels=design)  
fit2 = contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2)

iClustdifflistdir = paste(outputdir, &quot;/difflists_iClust_UCAM_only&quot;, sep=&quot;&quot;)
if(!file.exists(iClustdifflistdir))
    dir.create(iClustdifflistdir)

for(thiscontrast in contrasts)
{
  thisres = topTable(fit2, coef=thiscontrast, adjust=&quot;BH&quot;, number=999)

  thisname= gsub(&quot;-&quot;, &quot;-vs-&quot;, thiscontrast)
  thislabel = strsplit(thiscontrast, &quot;-&quot;)[[1]][1]

  difflistfile=paste(iClustdifflistdir, &quot;/difflist-iClust-&quot;,thisname, postfix, sep=&quot;&quot;)
  write.table(thisres, 
            file=difflistfile,
            quote=FALSE, sep=&quot;\t&quot;, row.names=TRUE, col.names=NA)
  print( paste(&quot;Diff genes with fdr&lt;0.05 for &quot;, thislabel, &quot; vs DCIS: &quot;,
               sum(thisres$adj.P.Val &lt; 0.05), sep=&quot;&quot;))

}
</code></pre>

<pre><code>## [1] &quot;Diff genes with fdr&lt;0.05 for iClust10 vs DCIS: 73&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust4 vs DCIS: 0&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust2 vs DCIS: 9&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust1 vs DCIS: 20&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust3 vs DCIS: 0&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust5 vs DCIS: 1&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust9 vs DCIS: 11&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust6 vs DCIS: 10&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust7 vs DCIS: 1&quot;
## [1] &quot;Diff genes with fdr&lt;0.05 for iClust8 vs DCIS: 17&quot;
</code></pre>

<p><br>
<br></p>

<pre><code class="r">sessionInfo()
</code></pre>

<pre><code>R version 3.1.1 (2014-07-10)
Platform: x86_64-apple-darwin10.8.0 (64-bit)

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] sva_3.12.0            genefilter_1.48.1     mgcv_1.8-6           
 [4] nlme_3.1-120          xtable_1.7-4          RColorBrewer_1.1-2   
 [7] knitr_1.10.5          AgiMicroRna_2.16.0    affycoretools_1.38.0 
[10] GO.db_3.0.0           RSQLite_1.0.0         DBI_0.3.1            
[13] AnnotationDbi_1.28.2  GenomeInfoDb_1.2.5    IRanges_2.0.1        
[16] S4Vectors_0.4.0       preprocessCore_1.28.0 affy_1.44.0          
[19] limma_3.22.7          Biobase_2.26.0        BiocGenerics_0.12.1  
[22] plyr_1.8.2           

loaded via a namespace (and not attached):
 [1] acepack_1.3-3.3           affyio_1.34.0            
 [3] annaffy_1.38.0            annotate_1.44.0          
 [5] AnnotationForge_1.8.2     base64enc_0.1-2          
 [7] BatchJobs_1.6             BBmisc_1.9               
 [9] BiocInstaller_1.16.5      BiocParallel_1.0.3       
[11] biomaRt_2.22.0            Biostrings_2.34.1        
[13] biovizBase_1.14.1         bit_1.1-12               
[15] bitops_1.0-6              brew_1.0-6               
[17] BSgenome_1.34.1           Category_2.32.0          
[19] caTools_1.17.1            checkmate_1.5.3          
[21] cluster_2.0.1             codetools_0.2-11         
[23] colorspace_1.2-6          DESeq2_1.6.3             
[25] dichromat_2.0-0           digest_0.6.8             
[27] edgeR_3.8.6               evaluate_0.7             
[29] fail_1.2                  ff_2.2-13                
[31] foreach_1.4.2             foreign_0.8-63           
[33] formatR_1.2               Formula_1.2-1            
[35] gcrma_2.38.0              gdata_2.16.1             
[37] geneplotter_1.44.0        GenomicAlignments_1.2.2  
[39] GenomicFeatures_1.18.7    GenomicRanges_1.18.4     
[41] GGally_0.5.0              ggbio_1.14.0             
[43] ggplot2_1.0.1             GOstats_2.32.0           
[45] gplots_2.17.0             graph_1.44.1             
[47] grid_3.1.1                gridExtra_0.9.1          
[49] GSEABase_1.28.0           gtable_0.1.2             
[51] gtools_3.4.2              Hmisc_3.16-0             
[53] hwriter_1.3.2             iterators_1.0.7          
[55] KernSmooth_2.23-14        lattice_0.20-31          
[57] latticeExtra_0.6-26       locfit_1.5-9.1           
[59] magrittr_1.5              markdown_0.7.7           
[61] MASS_7.3-40               Matrix_1.2-0             
[63] mime_0.3                  munsell_0.4.2            
[65] nnet_7.3-9                oligoClasses_1.28.0      
[67] OrganismDbi_1.8.1         PFAM.db_3.0.0            
[69] proto_0.3-10              R.methodsS3_1.7.0        
[71] R.oo_1.19.0               R.utils_2.0.2            
[73] RBGL_1.42.0               Rcpp_0.11.6              
[75] RcppArmadillo_0.5.100.1.0 RCurl_1.95-4.6           
[77] ReportingTools_2.6.0      reshape_0.8.5            
[79] reshape2_1.4.1            rpart_4.1-9              
[81] Rsamtools_1.18.3          rtracklayer_1.26.3       
[83] scales_0.2.4              sendmailR_1.2-1          
[85] splines_3.1.1             stringi_0.4-1            
[87] stringr_1.0.0             survival_2.38-1          
[89] tools_3.1.1               VariantAnnotation_1.12.9 
[91] XML_3.98-1.1              XVector_0.6.0            
[93] zlibbioc_1.12.0          
</code></pre>

<p>generation ended 2015-06-23 18:38:16. Time spent 0 minutes .</p>

</body>

</html>
